cli/src/semgrep/commands/ci.py

```diff
@@ -219,7 +219,16 @@ def ci(
         output_format=output_format,
     )
 
-    state.metrics.configure(metrics, metrics_legacy)
+    # Metrics are always on for CI scans. It doesn't make sense for them to
+    # be off, since CI scans have to connect to the Cloud Platform
+    # (For reviewers: is this explanation below clear?)
+    if metrics == MetricsState.OFF or metrics_legacy == MetricsState.OFF:
+        raise click.BadParameter(
+            "--metrics cannot be turned off for ci scans since ci scans require access to the Cloud Platform"
+        )
+    metrics = MetricsState.ON
```

Comment:
This change looks like it achieves the goal of getting metrics to on, but is there (also?) a more straightforward change we can make such that cloud platform initiated scans are already setting metrics to "ON" in the first place?  I worry we're generating an exception and potentially worrisome warning message on every app generated scan if not, but don't know enough to know where that change to default behavior is needed.

Reply:
If the specific concern is the exception, we don't have to disallow `metrics=off` for ci scans. I added this error because I remember in a previous discussion that we assumed `semgrep ci --metrics=off` was always a confusion (we send info to the app, even if not exactly the metrics). That being said, I'm perfectly happy to not have the warning and follow AUTO behavior.

Reply:
In the current case, a scan that uses CI but does not use registry will not use CLI metrics.  We want to change that behavior in general such that by default such a scan would use CLI metrics.  We don't consider CI without registry as an unusual setup, so I found handling it via exception unintuitive.

For example, could we set this at metrics configure time, e.g. in [metrics.py:configure(...)](https://github.com/returntocorp/semgrep/blob/60b623e4f66265599f0623622a1e1f7e34487972/cli/src/semgrep/metrics.py#L227).  If command is CI we do not check for registry usage before setting metrics to on by default.  But I am definitely not the expert on this code base, so will defer to your expertise.

I don't have enough context to know how we should handle the other question -- how to handle potentially unexpected / unintuitive setting of command CI with user-specified --metrics=off.  This is a particularly complicated question for users who are logged in to Cloud Platform.